aws/asg/asg.go

```diff
@@ -310,10 +311,38 @@ func (s *ASG) detach(asgc aws.ASGAPI) error {
 		if err != nil {
 			return err
 		}
+
+		// spin here and keep checking to see if all the instances are detached before returning
```

Comment:
This should probably be outside in a different Step splitting the `CleanUp` states in two. 

For now we could have it inside, this state though I would like to have limits on the number of retries (lets say 10 times). Given this is a serial operation and many services could have ALBs this could take a while and lambdas time limit is 15 mins.